src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregationManager.cs

```diff
@@ -130,13 +130,12 @@ private void PublishedInstrument(Instrument instrument, MeterListener _)
             {
                 _beginInstrumentMeasurements(instrument);
 
-                if (!_instruments.ContainsKey(instrument))
+                if (_instruments.TryAdd(instrument, true))
                 {
                     // This has side effects that prompt MeasurementsCompleted
                     // to be called if this is called multiple times on an
                     // instrument in a shared MetricsEventSource.
                     _listener.EnableMeasurementEvents(instrument, state);
-                    _instruments.Add(instrument, true);
```

Comment:
I am not that confident for this change CC @kkeirstead

Reply:
Assuming you're referring to the order of operations changing (the instrument getting added before `EnableMeasurementEvents`), that shouldn't be an issue - the side effects mentioned in the comment just explain why we can't call that method on the same instrument more than once. The relevant tests for this functionality are [here](https://github.com/kkeirstead/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricEventSourceTests.cs) - although I'm not certain whether `TryAdd` is available here

Reply:
> Assuming you're referring to the order of operations changing

Yes, that is what I was worried about 

> although I'm not certain whether TryAdd is available here

Build failed with error `CS1061: 'Dictionary<Instrument, bool>' does not contain a definition for 'TryAdd' and no accessible extension method 'TryAdd' accepting a first argument of type 'Dictionary<Instrument, bool>' could be found (are you missing a using directive or an assembly reference?)` 
Apparently it builds on .Net version that doesn't have the `Dictionary.TryAdd()`, will revert this change